search-app/src/cmr/search/services/query_execution/facets/hierarchical_v2_facets.clj

```diff
@@ -233,6 +234,15 @@
         links (generate-links-fn value has-siblings? children-values-to-remove)]
     (v2h/generate-hierarchical-filter-node value count links sub-facets field)))
 
+(defn- filter-out-accounted-for-facets
+  "Iterates over the new-facets and only keeps the leaf node (detailed-variable or short-name)
+  that doesn't already exists in the hierarchy that is represented by v2-node."
+  [sub-facets subfield new-facets]
+  (seq
+   (util/remove-nils-empty-maps-seqs
+    (map #(when-not (last-facet-accounted-for? sub-facets subfield (:title %)) %)
+         (:children new-facets)))))
```

Comment:
This is slightly confusing, but since you pass in `(:children new-facets)` do we skip a layer?
Because last-facet-accounded-for also starts with  `(map #(last-facet-accounted-for? % subfield title) (:children sub-facets)))`

Reply:
No.  new-facets is a structure that looks like what is listed below where we want to iterate over the children - the children are the list of facets we want to check to see if they were already included. 
{:title "Detailed_variable",
       :type :group,
       :applied true,
       :has_children true,
       :children [{:title "Snow Water Equivalent",
                   :type :filter,
                   :applied false,
                   :count 1,
                   :links {:apply "http:"},
                   :has_children false,
                   :field :detailed-variable}
                  {:title "Rain",
                   :type :filter,
                   :applied false,
                   :count 1,
                   :links {:apply "http:"},
                   :has_children false,
                   :field :detailed-variable}]}

The sub-facets that are passed in are a different created facet structure from previous processing.  We iterate over this one as well to get to the leaf nodes to see if any of the children from new-facets match the leaf nodes in the already created facet structure.